### DISCUSSION TITLE
Ograniczenie dostępu dla kontrybutora do AzureFiles

### DISCUSSION THREAD

**Discussion by chomamirek** [AzureCommunityPL/group#266] [Pomoc] [answered]
Witam
Posiadam skonfigurowany DFS z AzureFiles. 
Wszystko ładnie działa, uzytkownicy podłaczeni do udostępnionego zasobu mają dostęp przyznawany przez AD.
Problem jest taki iż te wszystkie pliki z serwera widać poprzez Azure Portal. 
Do mojej subskrypcji jest przydzielonych mnóstwo Kontrybutorów którzy nie powinni widzieć plików na Azure Files.
Czy jest jakiś sposób aby ograniczyć dostęp do zasobu dla kontrybutorów?

**Comment by KamilWiecek** [comment, marked as answer]
1. Nie nadużywać roli Contributor. 
2. Zaimplementować Azure PIM.
3. Stosować resource-specific roles. Na przykład: Network Contributor.
2. Utworzenie custom roli zawierającej notDataAction, które wywnioskujesz z wbudowanych ról, np. [Storage File Data SMB Share Elevated Contributor](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-file-data-smb-share-elevated-contributor)

**Reply by chomamirek** [reply]
Niestety do substrypcji musza byc przypisani Kontrybutorzy więc ta rola musi zostać. 
PIM mam zaimplementowany i działa ładnie jeśli chodzi aktywację roli Kontrybutor. 
Ale jeśli ktoś już aktywuje tą rolę to niestety ma dostęp do AzureFiles.
Custom Role niestety nie działa bo nie mogę  przypisać do Deny Assingments tak aby aby kontrybutorzy nie mieli dostępu. 
Microsoft sugeruje abym skorzystał z BluePrint ale niestety nie podają jak to zaimplementować.

**Reply by KamilWiecek** [reply]
Możesz:
- zrobić custom role, na podstawie obecnej roli Contributor, którą nazwiesz MyOwnContributor
- dodać notDataAction jw.
- zmienić role assignmenty z dawnego Contributor na MyOwnContributor

Zwróć uwagę na to, że chcesz użyc Azure Blueprinta(z którego za chwilę [będzie się trzeba migrować](https://github.com/AzureCommunityPL/group/discussions/209)) aby obejść problem, który wynika z nadania zbyt wysokich uprawnień wyżej w hierarchii. 

Rezultat będzie ten sam - obecny Contributor nie zobaczy plików w AzureFiles, ale:
- masz dodatkowy byt, którego deployment as a code to tragedia,
- zaciągasz gwarantowany dług technologiczny (migracja z Az BP).

**Reply by chomamirek** [reply]
Testuje od wczoraj i niestety wciąz MyOwnContributor widzi pliki w Azure Share File.

**Reply by KamilWiecek** [reply]
Jeżeli nie masz załączonego **Identity-based access for file shares** to musisz użyć notActions. 
Dopracuj sobie to co wrzucam Ci poniżej i będziesz w domu :)

```json
{
    "properties": {
        "roleName": "MyContributor",
        "description": "",
        "assignableScopes": [
            "XXX"
        ],
        "permissions": [
            {
                "actions": [
                    "*"
                ],
                "notActions": [
                    "Microsoft.Authorization/*/Delete",
                    "Microsoft.Authorization/*/Write",
                    "Microsoft.Authorization/elevateAccess/Action",
                    "Microsoft.Blueprint/blueprintAssignments/write",
                    "Microsoft.Blueprint/blueprintAssignments/delete",
                    "Microsoft.Compute/galleries/share/action",
                    "Microsoft.Storage/storageAccounts/fileServices/shares/action",
                    "Microsoft.Storage/storageAccounts/fileServices/read",
                    "Microsoft.Storage/storageAccounts/fileServices/write",
                    "Microsoft.Storage/storageAccounts/fileServices/shares/delete",
                    "Microsoft.Storage/storageAccounts/fileServices/shares/read",
                    "Microsoft.Storage/storageAccounts/fileServices/shares/lease/action",
                    "Microsoft.Storage/storageAccounts/fileServices/shares/write"
                ],
                "dataActions": [],
                "notDataActions": [
                    "Microsoft.Storage/storageAccounts/fileServices/fileshares/files/read",
                    "Microsoft.Storage/storageAccounts/fileServices/fileshares/files/write",
                    "Microsoft.Storage/storageAccounts/fileServices/fileshares/files/delete",
                    "Microsoft.Storage/storageAccounts/fileServices/fileshares/files/modifypermissions/action",
                    "Microsoft.Storage/storageAccounts/fileServices/fileshares/files/actassuperuser/action"
                ]
            }
        ]
    }
}
```

**Reply by chomamirek** [reply]
@kwiecek 
Zadziałało tak jak chciałem, aczkolwiek trzeba poczekać trochę az rola się zastosuje.
Wielkie dzięki za pomoc.
